.github/workflows/ci-workflow.yml

```diff
@@ -0,0 +1,74 @@
+name: native-metrics CI
+
+on:
+  push:
+    branches:
+    - open-source-main
+  pull_request:
+    branches:
+    - '**'
```

Comment:
what does this do? seems unnecessary...

Reply:
It seems in my testing that it triggers the action on the branch you are Pull requesting from.  If I changed the branches to 'open-source-main' for example, the action would not run.

Reply:
still not sure i understand.